### ISSUE TITLE
Consider alternatives to GitHub Pages

### ISSUE THREAD

**Issue by handrews** [OAI/OpenAPI-Specification#4153]
GitHub Pages has several very problematic limitations for a project focused on HTTP standards:
* It does not allow setting `Content-Type` except via file extension, which is not how our schemas work
* It does not support HTTP redirects, only HTML-driven refresh-redirects

The first two are essential for us, as we want to have our JSON Schemas served as JSON, and we want to have a `latest` resource that uses redirects instead of serving a schema with a mismatched `$id` (this helps prevent people from thinking that the `latest` URI can be used in `$ref`s, etc.).

Netlify seems to be the most popular suggestion on our Slack, with @jmertic having seen it used for similar purposes and at least one other person recommending it.  I've also seen Cloudflare Pages used.  Both can run from GitHub repos, and I'm pretty sure both can solve both of our main problems.  The distinction might be which has the most services available through any sort of open source foundation-oriented plan (I believe both have or at least once had such plans), which I have not investigated at all.

**Comment by handrews** [comment]
Also, jekyll and liquid frustrate everyone endlessly AFAICT.  We'd have much better sites in a better system.

**Comment by lornajane** [comment]
I'm very much in favour of choosing a different/better platform. I'd like to note that learn.openapis.org does also use Jekyll.

**Comment by handrews** [comment]
@lornajane yeah spec.openapis.org, learn.openapis.org, and tools.openapis.org are all GitHub Pages/Jekyll/Liquid.  While the frustration is most acute with the spec site, it's a problem for the others as well, just in terms of the flexibility of layout and who is able to work on them.

**Comment by SensibleWood** [comment]
@handrews @lornajane tools.openapis.org doesn't use Jekyll, it uses Eleventy with Nunjucks. Really easy, JavaScript based, removes the need for any Ruby based hoo-ha.

Not especially wedded to GitHub Pages. Used Netlify before, seems good especially given supports a promotional model which is always a good thing IMHO.

**Comment by handrews** [comment]
@SensibleWood oh good to know!  That explains why the tools site can have that nice panel layout.  It never occurred to me to check for some reason.

**Comment by baywet** [comment]
This was discussed about during the TDC meeting today.
We really want to improve the main website, and possibly align the tech stacks for the main website, the specs, tools, learn to reduce the friction here.
Additionally, if we could solve the technical pain points that @handrews will share, it'd be perfect.
We're considering switching both the static site generation AND the hosting platform.
